flask/app/models.py

```diff
@@ -450,17 +459,30 @@ class Variant(db.Model):
     polyphen_score: int = db.Column(db.Float, nullable=True)
     cadd_score: int = db.Column(db.Float, nullable=True)
     gnomad_af: int = db.Column(db.Float, nullable=True)
+    gene = db.relationship("Gene", back_populates="variant")
+    __table_args__ = (db.UniqueConstraint("variant_id", "analysis_id"),)
 
 
-class AnalyzedVariant(db.Model):
+@dataclass
+class Genotype(db.Model):
     variant_id: int = db.Column(
         db.Integer, db.ForeignKey("variant.variant_id"), primary_key=True
     )
     analysis_id: int = db.Column(
-        db.Integer, db.ForeignKey("analysis.analysis_id"), primary_key=True
+        db.Integer,
+        db.ForeignKey("analysis.analysis_id"),
+        primary_key=True,
     )
-    analyses = db.relationship("Analysis", backref="variant")
-    variant = db.relationship("Variant", backref="analysis")
+    dataset_id: int = db.Column(
+        db.Integer,
+        db.ForeignKey("dataset.dataset_id"),
+        primary_key=True,
+    )
+
+    variants = db.relationship("Variant", backref="variant_dataset_analysis")
+    analyses = db.relationship("Analysis", backref="variant_dataset_analysis")
+    datasets = db.relationship("Dataset", backref="variant_dataset_analysis")
```

Comment:
I'm confused about the naming of these relationships and also they're considering the foreign keys independently rather than in conjunction.

Reply:
i used the documentation for reference on how to create an association table - https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#association-object, hopefully that clears up the relationships?

as for the constraint i didn't explicitly add one as the three primary keys are considered a composite primary key (see lines 105 to 106 in the alembic migration)

Reply:
Did you have to write the composite constraint in the migration manually or was Alembic able to figure that out automatically?

Reply:
Alembic figured it out automatically

Reply:
Looking into your comment below re: redundancy of the `variant_id` unique constraint it looks like Alembic figured out the composite primary key, but I must have manually added the unique constraint (which would also be redundant I believe).